src/core/Microsoft.PowerToys.Settings.UI.Runner/MainWindow.xaml.cs

```diff
@@ -67,17 +67,14 @@ private void WindowsXamlHost_ChildChanged(object sender, EventArgs e)
                 {
                     if (ShellPage.ShellHandler.IPCResponseHandleList != null)
                     {
-                        try
+                        var success = JsonObject.TryParse(msg, out JsonObject json);
+                        if (success)
                         {
-                            JsonObject json = JsonObject.Parse(msg);
                             foreach (Action<JsonObject> handle in ShellPage.ShellHandler.IPCResponseHandleList)
                             {
                                 handle(json);
                             }
                         }
```

Comment:
I think we want to log an error or maybe even throw if we are getting IPC messages formatted incorrectly

Reply:
Currently it would swallow all the exceptions, and from looking at the code this https://github.com/microsoft/PowerToys/blob/5b86885cd1c0c3314d0a0a5b78a081509c982e5b/src/core/Microsoft.PowerToys.Settings.UI/Views/GeneralPage.xaml.cs#L49-L75 was the only handler I could find which already swallows its own exceptions. So I thought maybe the only possible exceptions were from JsonObject.Parse, but I agree on logging them at least. Should I log -and- throw here or just log them?

Reply:
My preference would be to use the tryparse, and then log the error.    We also may want to consider adding a debugger.break in debug when we log the error, but that is a seperate issue.

Reply:
@ryanbodrug-microsoft okay I added that in - let me know what you think.